h2/src/test/org/h2/test/testScript.sql

```diff
@@ -2,6 +2,83 @@
 -- and the EPL 1.0 (http://h2database.com/html/license.html).
 -- Initial Developer: H2 Group
 --
+-- INT-2835 test case
+create table application(id varchar(50) not null, appName varchar(40), constraint app_pk primary key (id));
+> ok
+
+create table pull_request(id varchar(50) not null, appId varchar(50) not null, prId int not null, constraint pr_pk primary key (id), constraint app_fk foreign key(appId) references application(id), constraint source_control_pull_request_app_component_pull_request_uk unique (appId, prId));
+> ok
```

Comment:
the above tables are setup in a similar structure to what we have in IQ; i.e. foreign key to application table which is also used in the unique constraint.